core/src/test/java/dev/keva/core/server/AbstractServerTest.java

```diff
@@ -827,6 +830,131 @@ void setrange() {
+                jedis.zadd("zset", 1.0, "val", new ZAddParams().xx().nx());
+            });
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddSingleWithNxAndGtErrs() {
+        try {
+            assertThrows(JedisDataException.class, () -> {
+                jedis.zadd("zset", 1.0, "val", new ZAddParams().gt().nx());
+            });
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddSingleWithNxAndLtErrs() {
+        try {
+            assertThrows(JedisDataException.class, () -> {
+                jedis.zadd("zset", 1.0, "val", new ZAddParams().lt().nx());
+            });
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddSingleWithGtAndLtErrs() {
+        try {
+            assertThrows(JedisDataException.class, () -> {
+                jedis.zadd("zset", 1.0, "val", new ZAddParams().lt().gt());
+            });
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddSingleWithoutOptions() {
+        try {
+            var result = jedis.zadd("zset", 1.0, "val");
+            assertEquals(1, result);
+
+            result = jedis.zadd("zset", 1.0, "val");
+            assertEquals(0, result);
+        } catch (Exception e) {
+            fail(e);
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddMultipleWithoutOptions() {
+        try {
+            Map<String, Double> members = new HashMap<>();
+            int numMembers = 100;
+            for(int i=0; i<numMembers; ++i) {
+                members.put(Integer.toString(i), (double) i);
+            }
+            var result = jedis.zadd("zset", members);
+            assertEquals(numMembers, result);
+
+            result = jedis.zadd("zset", members);
+            assertEquals(0, result);
+        } catch (Exception e) {
+            fail(e);
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zaddCh() {
+        try {
+            var result = jedis.zadd("zset", 1.0, "mem", new ZAddParams().ch());
+            assertEquals(1, result);
+
+            result = jedis.zadd("zset", 1.0, "mem", new ZAddParams().ch());
+            assertEquals(0, result);
+
+            result = jedis.zadd("zset", 2.0, "mem", new ZAddParams().ch());
+            assertEquals(1, result);
+        } catch (Exception e) {
+            fail(e);
+        } finally {
+            jedis.del("zset");
+        }
+    }
+
+    @Test
+    void zscoreNonExistingKey() {
+        val result = jedis.zscore("key", "mem");
+        assertNull(result);
+    }
+
+    @Test
+    void zscoreNonExistingMember() {
+        try {
+            jedis.zadd("zset", 1.0, "mem");
+            val result = jedis.zscore("zset", "foo");
+            assertNull(result);
+        } finally {
+            jedis.del("zset");
```

Comment:
Hi @the123saurav, we might don't need this `finally` block because we already reset the database after every `@Test` method

Ref: https://github.com/keva-dev/keva/blob/master/core/src/test/java/dev/keva/core/server/KevaServerTest.java#L54-L57

Reply:
Ohh missed it completely, will remove.